File: Wms.Api/Dto/StockReservation/StockReservationCreateUpdate/StockReservationCreateUpdateDto.cs
using Wms.Api.Model;

namespace Wms.Api.Dto.StockReservation.StockReservationCreateUpdate
{
    public class StockReservationCreateUpdateDto
    { 
        public required string Number { get; set; }
        public Guid WarehouseId { get; set; }
        public DateTime ReservedAt { get; set; }
        public DateTime ExpiresAt { get; set; }
        public required string Status { get; set; }
        public string? CancellationRemark { get; set; }
        public string? CancellationRequestedBy { get; set; }
        public DateTime? CancellationRequestedAt { get; set; }
        public string? CancellationApprovedBy { get; set; }
        public DateTime? CancellationApprovedAt { get; set; }
        public ICollection<StockReservationItemCreateUpdateDto>? StockReservationItems { get; set; }
    }
}

File: Frontend/src/interfaces/v12/stockReservation/stockReservationCreateUpdate/stockReservationCreateUpdateDto.ts
import { ReservationStatusEnum } from "interfaces/enums/GlobalEnums";
import { guid } from "types/guid";

export interface StockReservationCreateUpdateDto {
  number?: string | null;
  warehouseId: guid;
  reservedAt: string;
  expiresAt: string;
  status: ReservationStatusEnum;
  cancellationRemark?: string | null;
  cancellationRequestedBy?: string | null;
  cancellationRequestedAt?: string | null;
  cancellationApprovedBy?: string | null;
  cancellationApprovedAt?: string | null;
  stockReservationItems: StockReservationItemCreateUpdateDto[] | null;
}

export interface StockReservationItemCreateUpdateDto {
  productId: string;
  quantity: number;
}
